stl/inc/format

```diff
@@ -1980,14 +2157,37 @@ _NODISCARD _OutputIt _Write(
         throw format_error("string cannot be localized");
     }
 
-    auto _Printed_size = static_cast<int>(_Value.size());
+    if (_Specs._Precision < 0 && _Specs._Width <= 0) {
+        return _Write(_STD move(_Out), _Value);
+    }
+
+    auto _Pos                  = _Value.data();

```

Comment:
Everything from here to 2187 is independent of `_OutIt`, and should be extracted into a separate function template that returns `_Pos` and `_Estimated_width`.